modules/WebcastPolls/resources/webcastPolls.js

```diff
@@ -237,52 +275,178 @@
                 _this.view.parent = _this;
             }
 
-            if (!_this.cuePointsManager) {
-                // cue points manager used to monitor and notify when relevant cue points reached (polls status, results).
-                _this.cuePointsManager = new mw.webcast.CuePointsManager(_this.getPlayer(), function () {
-                }, "webcastPolls_CuePointsManager");
 
-                _this.cuePointsManager.registerMonitoredCuepointTypes(['poll-data'],function(cuepoints)
-                {
-                   for(var i = 0;i< cuepoints.length;i++)
-                   {
-
-                       try {
-                           var cuepoint = cuepoints[i];
-                           var cuepointContent = cuepoint.partnerData ? JSON.parse(cuepoint.partnerData) : null;
-                            var pollIdTokens = (cuepoint.tags || '').match(/id:([^, ]*)/);
-                           var pollId = pollIdTokens && pollIdTokens.length === 2 ? pollIdTokens[1] : null;
-
-                           if (cuepointContent && pollId)
-                           {
-                               var pollContent = cuepointContent.text;
-
-                               if (pollId && pollContent) {
-                                   _this.log("updated content of poll with id '" + pollId + "'");
-                                   if(_this.globals.pollsContentMapping[pollId]) {
-                                        $.extend(_this.globals.pollsContentMapping[pollId], pollContent);
-                                   }else {
-                                       _this.globals.pollsContentMapping[pollId] = pollContent;
-                                   }
-                               }
-                           }
-
-                       }catch(e)
-                       {
-                           _this.log("ERROR while tring to extract poll information with error " + e);
-                       }
-
-                   }
-                });
+            if(_this.embedPlayer.isLive()) {
+				//Live mode - use push notifications engine
+				_this.pushCuePointsManager = mw.KPushCuePointsManager.getInstance(this.embedPlayer);
+				_this.pushCuePointsManager.registerToNotification(_this.polls_push_notification, _this.getUserID(),
+					this.cuePointLoaded, this.cuePointReached, this, "polls");
+				//TODO - Eitan - detach
+			}
+            // }else{
+                // VOD mode
+                if (!_this.cuePointsManager) {
+                    // cue points manager used to monitor and notify when relevant cue points reached (polls status, results).
+                    _this.cuePointsManager = new mw.webcast.CuePointsManager(_this.getPlayer(), function () {
+                    }, "webcastPolls_CuePointsManager");
+
+                    _this.cuePointsManager.registerMonitoredCuepointTypes(['poll-data'],function(cuepoints)
+                    {
+						// console.log(">>>>> @@", "registerMonitoredCuepointTypes" , cuepoints);
```

Comment:
please delete code that is not relevant anymore . (you can always look at git for history)